### PR TITLE
Add sticky footer for logging consent visibility

### PR DESCRIPTION
**Pull request by Kokobe**
* Adds a sticky footer for the WelcomeScreen so that the logging consent and privacy policy does not get hidden below the Benefits promo. 
* Also: Removes `SettingsDataModel` from the `WelcomeFlowNavModel` because it's not being used anymore. 

Before:
Screen Shot 2023-03-11 at 4 43 37 PM

After:

https://user-images.githubusercontent.com/20916043/224518137-f567e677-d2a7-43a2-bb63-65e404f17bc1.mov

On Samsung:

https://user-images.githubusercontent.com/20916043/224518305-f8369099-ad4f-43a5-b139-be5a947db05a.mp4

### Pointers to help with review
* https://www.figma.com/file/dylvDPB69JG9U1rzZ3HQ52/Android-Premium?node-id=0%3A1&t=UMXd6c7grlKOC2Tb-0

### How was this tested?
- [x] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [ ] I added automated tests
- [x] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [x] I checked that `@Previews` still work in Android Studio

### CODE REVIEW

File: screenshotTests/src/main/assets/golden/com.neeva.app.welcomeflow.WelcomeFlowScreenshotTests_welcomeScreen_Dark_Preview.png
Comment (dan-alcantara-neeva) [inline screenshotTests/src/main/assets/golden/com.neeva.app.welcomeflow.WelcomeFlowScreenshotTests_welcomeScreen_Dark_Preview.png:1]: Fix the padding on the right of the checkbox

File: app/src/androidTest/java/com/neeva/app/firstrun/FirstRunStartupTest.kt [hunk 1]
```diff
@@ -78,11 +77,18 @@ class FirstRunStartupTest : BaseBrowserTest() {
 
     override fun setUp() {
         super.setUp()
-
         testAuthenticatedApolloWrapper =
             authenticatedApolloWrapper as TestAuthenticatedApolloWrapper
     }
 
+    private fun currentlyOnDestination(
```
Comment (dan-alcantara-neeva) [inline app/src/androidTest/java/com/neeva/app/firstrun/FirstRunStartupTest.kt:84]: isCurrentlyAtDestination?

**Review by dan-alcantara-neeva — CHANGES_REQUESTED** [review]

**Comment by hwiatneeva** [comment]
from design: LGTM - if we have decided to go with and implemented the sticky foooter, we should go with it. Although the amount of the content in a small screen bugs me, let's push it and move on. adding this comment since it seems an okay from design can add a tiny support. Thanks. @Kokobe @dan-alcantara-neeva

**Comment by Kokobe** [comment]
@hwiatneeva the small screen example is an extreme but it's an example of what could happen if the user uses a larger font/display size. As a consequence of having a lot of text on the screen (and having a sticky footer), cases like that tend to happen a bit more easily -- but I believe that the sticky footer is necessary information on the screen that should not be hidden

**Review by dan-alcantara-neeva — APPROVED** [review]

**Pull request merged**